### ISSUE TITLE
"assignment-from-none" false positive on generator

### ISSUE THREAD

**Issue by CrepeGoat** [PyCQA/pylint#3519]
### Steps to reproduce
1. Create a new .py file with the following contents:
```python
def paused_iter(iterable):
    """Iterate an iterable, yielding the previous item using `i.send(True)`."""
    iterator = iter(iterable)

    while True:
        try:
            value = next(iterator)
        except StopIteration:
            return

        # Continue to yield the same item until `next(i)` or `i.send(False)`
        while (yield value):
            pass


ITERATOR = paused_iter(range(3))
```
2. Run `pylint` on this file.

### Current behavior
Pylint reports the following error:
```
/path/to/file.py:15:0: E1128: Assigning result of a function call, where the function returns None (assignment-from-none)
```

### Expected behavior
The above error does not apply to the above valid code since the function that "returns None" is a generator. Thus, I expect that no such error should be reported.

### pylint --version output
```
pylint 2.4.4
astroid 2.3.3
Python 3.7.5 (default, Nov  7 2019, 10:50:52) 
[GCC 8.3.0]
```

**Comment by PCManticore** [comment]
Thanks for the report!
